### DISCUSSION TITLE
python binding

### DISCUSSION THREAD

**Discussion by TheforgottenDreamer** [jrouwe/JoltPhysics#1090] [Ideas]
hi, so I am someone who's working with pygame and moderngl and was seeking an API for physics. I am wondering if the devs will be making a Python binding

**Comment by jrouwe** [comment]
Hello,

I'd love to have python bindings, but as it stands I don't have time to create one. I think it should not be hard if you use one of the projects that export C bindings for Jolt: [here](https://github.com/michal-z/zig-gamedev/tree/main/libs/zphysics/libs) and [here](https://github.com/amerkoleci/JoltPhysicsSharp/tree/main/src/joltc).

**Reply by TheforgottenDreamer** [reply]
I'm planning to use Swig to maybe make a binding

**Reply by MikelAyani** [reply]
Hi! I am very interested in using JoltPhysics with Python, any progress?

**Comment by Rohan-Gautam** [comment]
Pymunk: A popular 2D physics engine known for its ease of use, performance, and extensive documentation. It integrates well with Pygame for creating realistic 2D physics simulations http://www.pymunk.org/.


Box2D: Another well-regarded 2D physics library, offering a variety of shapes, joints, and collision detection methods. It can be used with Pygame to create complex physical interactions https://box2d.org/documentation/.


Chipmunk/Chipmunk2D: The C++ foundation of Pymunk, providing a lower-level API for more granular control over physics simulations. It can be integrated with Pygame for advanced physics needs, but requires more C++ knowledge https://github.com/slembcke/Chipmunk2D.

**Reply by TheforgottenDreamer** [reply]
Yeah but i am using pygame for 3d stuff and pybullet is quite strange to convert to glm

**Comment by MikelAyani** [comment]
I am willing to give this a try, anyone that wants to help?

**Reply by parsany** [reply]
Yea, I'll start working on it following the next 3 weeks
busy with exams currently

**Reply by MikelAyani** [reply]
Great! Let me know how do you want to proceed and we try to colaborate if you want.

**Reply by hiraz01** [reply]
Did you guys ever get going on this? I think this could bring in the Reinforcement Learning community and perhaps enhance Jolt as a whole.

**Reply by hiraz01** [reply]
BTW. I am interested in helping.

**Reply by MikelAyani** [reply]
Can we move to a discord room and discuss how we do it together? I have personally started with it.

**Reply by gamedev44** [reply]
@MikelAyani 
i am interested in helping currently im extending the python3d engine

**Reply by MikelAyani** [reply]
Ok, let's create a Discord channel then and take it from there:
https://discord.gg/DWChywzw

**Reply by hiraz01** [reply]
I just joined the Discord channel. Apologies for the late response.
